### ISSUE TITLE
Discussion : do we need a windows installer ?

### ISSUE THREAD

**Issue by biconou** [Libresonic/libresonic#204]
Guidelines for Contributing
---------------------------

Libresonic development is a community project, and contributions are welcomed. Here are a few guidelines you should follow before submitting:

  1.  **License Acceptance** All contributions must be licensed as [GNU GPLv3](https://github.com/Libresonic/libresonic/blob/develop/LICENSE.txt) to be accepted. Use [`git commit --signoff`](https://jk.gs/git-commit.html) to acknowledge this.
  2.  **No Breakage** New features or changes to existing ones must not degrade the user experience. This means do not introduce bugs, remove functionality, or make large changes to existing themes/UI without prior discussion in an Issue.
  3.  **Coding standards** Language best-practices should be followed, comment generously, and avoid "clever" algorithms. Refactoring existing messes is great, but watch out for breakage.
  4.  **Be bold!** Without contributions, this project will vanish. If you just want to help out, try [submiting a patch](https://github.com/Libresonic/libresonic/issues?q=is%3Aissue+is%3Aopen+label%3Apatches-welcome) for an open Issue.
  5.  **Stay relevant** Issues or commentary that is off-topic or tangential to Libresonic development is subject to moderation. Questions should be focused on improving documentation to solve a problem. Visit [Reddit](https://www.reddit.com/r/libresonic) or [IRC](http://webchat.freenode.net?channels=%23libresonic) for community discussion.

**Comment by plastikman** [comment]
Now that the spring boot changes are merged it can be looked at.  Perhaps we should see how many people actually use the windows installer then make a decision?

The windows installer significantly lowers the barrier to entry, but will require testing at each release.  I am personally running libresonic under tomcat on windows 10 without any issues.  

Maybe it makes sense to just document the install steps to run tomcat on windows and be done with it?

**Comment by plastikman** [comment]
I can see that there is a requirement for https://www.ej-technologies.com/download/exe4j/files to make the installer, this is payware.  The installer will pop up a message complaining about the exe was created with a trial program unless its purchased. 

There is a request for an opensource project license [here](https://www.ej-technologies.com/buy/install4j/openSource/enter).

**Comment by biconou** [comment]
Personally, I think that installers in the traditional form will become outdated.
The concept of continuous delivery and appstores is the future. 
I think that Docker is part of this story. Linux packages (debian, rpm) is good too.
To deliver Libresonic on Windows machines perhaps there is a sort of windows store solution ? (I don't really know much about Microsoft as I definitely turned to Linux).

**Comment by zifnab06** [comment]
choco for windows, if you really care. Personally - release the war, make the war also work with java -jar libresonic.war, and if people need something more advanced they can do it themselves (ie, installing tomcat)

**Comment by biconou** [comment]
Do you mean https://chocolatey.org/ ?

**Comment by eizal** [comment]
I'm really much looking forward to this. For one year I was a subscriber to Subsonic. Because my use at the time was very scarce, I decided to to use the workarounds there after. Now that those aren't working anymore I looked into alternatives and found Libresonic.

I'm confused is this yet working on Windows or just Linux? But maybe this is not the right thread for that. Anyway I would be installing and using this as soon as possible on W10 platform. Cheers.

**Comment by biconou** [comment]
It works on Windows but we are not much focused on creating a user frendly installer for windows. 
You will have to deploy the war to a Tomcat.
Do you know how to do that ?

**Comment by eizal** [comment]
No. But I will try to find out. Unless you care to help a little?

**Comment by biconou** [comment]
I can help you but if you just wait for the 6.2 version, it will be easier for you as you'll no more need Tomcat.
What do you think ?

**Comment by eizal** [comment]
Thanks for the hint. I'll do the wait. :)

**Comment by Zabeus** [comment]
I'm also interested in Subsonic on Windows and will try building it today.
This looks relevant: https://github.com/Libresonic/libresonic/issues/37

**Comment by muff1nman** [comment]
I would follow the doc at INSTALL.md using tomcat. Unless you need something in 6.2 or would like to help out testing, use the stable 6.1 branch. Although the INSTALL.md file is not geared towards windows, the general idea should be clear. If it's not clear, let us know.

**Comment by sheis** [comment]
Also interested in learning to use this. Cheers.

**Comment by muff1nman** [comment]
I am actually thinking it would be best to remove all of the installer components from this repository and create separate ones based on developer support/interest. I can see the value in having a Windows installer but unfortunately I don't think we have enough man power to focus on all targets for every release. Furthermore every single installer has references to the to be deleted booter project so each needs updating for 6.2. Personally I'd like to see the 6.2 spring boot release not get delayed while we update all of the current install targets.

**Comment by biconou** [comment]
I agree with @muff1nman

**Comment by fozzedout** [comment]
As long as there are instructions for me to follow, I don't mind what installation method is used. I'm currently on Windows using an outdated docker version of Libresonic, as was easier to get up and running than dealing with java, tomcat and whatever else is needed to be set up first, so if a docker container is the way to go - I'm all ears

**Comment by barretts** [comment]
I've just recently (yesterday) gotten Libresonic to work on Windows by building the `libresonic-installer-windows` files and replacing the installed files of Subsonic. I like having a Windows service that starts up automatically and I never really have to think about it running. I don't have Tomcat already running on my system and the installation instructions available assume a working knowledge of that server. Running a docker instance is straight forward but soaks up a good deal more system resources. I think the setup I had for a long time was thanks to someone just posting a built executable online https://github.com/Libresonic/libresonic/issues/37.

Perhaps there is some exe4j alternative (NSIS, izpack) that would work. It would certainly make things easier to have a Windows installer available but if the installation instructions were more granular I think that would work. With answers such as `how can I setup libresonic to start with Windows?`

This is from my experience as a Windows user unfamiliar with Tomcat and migrating from Subsonic.
